app/models/integrations/app.rb

```diff
@@ -25,10 +26,21 @@ def fields
     params[:fields]
   end
 
+  # There is not way to get the account_id from the linear callback
+  # so we are using the generate_linear_token method to generate a token and encode it in the state parameter
+  def encode_state
+    generate_linear_token(Current.account.id)
+  end
+
   def action
     case params[:id]
     when 'slack'
       "#{params[:action]}&client_id=#{ENV.fetch('SLACK_CLIENT_ID', nil)}&redirect_uri=#{self.class.slack_integration_url}"
+    when 'linear'
+      base_url = "#{params[:action]}?response_type=code"
+      "#{base_url}&client_id=#{ENV.fetch('LINEAR_CLIENT_ID', nil)}" \
+        "&redirect_uri=#{self.class.linear_integration_url}" \
+        "&state=#{encode_state}&scope=read,write&prompt=consent"
```

Comment:
Let's move this to a separate fn called `build_linear_action`. We can do something like the following

```rb
def build_linear_action
    [
      "#{params[:action]}?response_type=code",
      "client_id=#{ENV['LINEAR_CLIENT_ID']}",
      "redirect_uri=#{self.class.linear_integration_url}",
      "state=#{encode_state}",
      'scope=read,write',
      'prompt=consent'
    ].join('&')
  end
  ```